src/Rings/mpolyquo-localizations.jl

```diff
@@ -2698,27 +2698,54 @@ function Base.:(==)(
 
   if _has_coefficient_map(f)
     if _has_coefficient_map(g) 
-      coefficient_map(f) == coefficient_map(g) || return false
+      return is_equal_as_morphism(coefficient_map(f), coefficient_map(g))
     else
-      coefficient_map(f) == identity_map(coefficient_ring(domain(f))) || return false
+      return is_trivial(coefficient_map(f))
     end
   elseif _has_coefficient_map(g)
     if _has_coefficient_map(f) 
-      coefficient_map(f) == coefficient_map(g) || return false
+      return is_equal_as_morphism(coefficient_map(f), coefficient_map(g))
     else
-      coefficient_map(g) == identity_map(coefficient_ring(domain(g))) || return false
+      is_trivial(coefficient_map(g))
     end
   end
 
-  return all(x->f(x) == g(x), gens(domain(f)))
+  return all(f(x) == g(x) for x in gens(domain(f)))
+end
+
+# TODO: Move to wherever `MapFromFunc` comes from ?
+function is_equal_as_morphism(f::MapFromFunc, g::MapFromFunc)
```

Comment:
you can move this function to `Misc/MoveToHecke.jl`, then we remember to move it to Hecke once we do another large moving upstream attempt.